cmd/agent/api/server.go

```diff
@@ -65,122 +64,66 @@ func StartServer(
 	invAgent inventoryagent.Component,
 	invHost inventoryhost.Component,
 ) error {
-	err := initializeTLS()
+	apiAddr, err := getIPCAddressPort()
 	if err != nil {
-		return fmt.Errorf("unable to initialize TLS: %v", err)
+		return fmt.Errorf("unable to get IPC address and port: %v", err)
 	}
 
-	// get the transport we're going to use under HTTP
-	listener, err = getListener()
-	if err != nil {
-		// we use the listener to handle commands for the Agent, there's
-		// no way we can recover from this error
-		return fmt.Errorf("Unable to create the api server: %v", err)
-	}
-
-	err = util.CreateAndSetAuthToken()
+	extraHosts := []string{apiAddr}
+	ipcHost, err := config.GetIPCAddress()
 	if err != nil {
 		return err
 	}
+	apiConfigPort := config.Datadog.GetInt("api_config_port")
+	apiConfigEnabled := apiConfigPort > 0
+	apiConfigHostPort := net.JoinHostPort(ipcHost, strconv.Itoa(apiConfigPort))
```

Comment:
We want a different ipcHost for this API so we can bind it to a different interface.

Reply:
Done